starcheck/plot.py

```diff
@@ -0,0 +1,327 @@
+                         fill=False)
+    ax.add_patch(box1)
+    b2w = 2600
+    box2 = plt.Rectangle((b2w, -b1hw), -4 + -2 * b2w, 2 * b1hw,
+                         fill=False)
+    ax.add_patch(box2)
+
+    ax.scatter([-2700, -2700, -2700, -2700, -2700],
+               [2400, 2100, 1800, 1500, 1200],
+               c='orange', edgecolors='orange',
+               s=symsize(np.array([10.0, 9.0, 8.0, 7.0, 6.0])))
+
+    [l.set_rotation(90) for l in ax.get_yticklabels()]
+    ax.grid(grid)
+    ax.set_ylabel("Zag (arcsec)")
+    ax.set_xlabel("Yag (arcsec)")
+
+    if quad_bound:
+        pix_range = np.linspace(-510, 510, 50)
+        minus_half_pix = -0.5 * np.ones_like(pix_range)
+        # plot the row = -0.5 line
+        yag, zag = pixels_to_yagzag(minus_half_pix, pix_range)
+        ax.plot(yag, zag, color='magenta', alpha=.4)
+        # plot the col = -0.5 line
+        yag, zag = pixels_to_yagzag(pix_range, minus_half_pix)
+        ax.plot(yag, zag, color='magenta', alpha=.4)
+
+    # plot starcheck catalog
+    if catalog is not None:
+        _plot_catalog_items(ax, catalog)
+    # plot field if present
+    if stars is not None:
+        if attitude is None:
+            raise ValueError("Must supply attitude to plot field stars")
+        _plot_field_stars(ax, stars, Quaternion.Quat(attitude),
+                          bad_stars=bad_stars, red_mag_lim=red_mag_lim)
+    if title is not None:
+        fig.suptitle(title, fontsize='small')
+    return fig
+
+
+def make_plots_for_obsid(obsid, ra, dec, roll, starcat_time, catalog, outdir, red_mag_lim=10.7):
+    """
+    Make standard starcheck plots for obsid and save as pngs with standard names.
+    Writes out to stars_{obsid}.png and star_view_{obsid}.png in supplied outdir.
+
+    :param obsid:  Obsid used for file names
+    :param ra: RA in degrees
+    :param dec: Dec in degrees
+    :param roll: Roll in degrees
+
+    :param catalog: list of dicts or other astropy.table compatible structure with conventional
+                    starcheck catalog parameters for a set of ACQ/BOT/GUI/FID/MON items.
+    :param outdir: output directory for png plot files
+    :param red_mag_lim: faint limit
+    """
+
+    # explicitly float convert these, as we may be receiving this from Perl passing strings
+    ra = float(ra)
+    dec = float(dec)
+    roll = float(roll)
+    # get the agasc field once and then use it for both plots that have stars
+    field_stars = agasc.get_agasc_cone(ra, dec,
+                                       radius=1.5,
+                                       date=DateTime(starcat_time).date)
+
+    # Mark bad stars from current limits in characteristics
+    bad_stars = ((field_stars['CLASS'] != 0) | (field_stars['MAG_ACA_ERR'] > 100) |
+                 (field_stars['POS_ERR'] > 3000) | (field_stars['ASPQ1'] > 0) |
+                 (field_stars['ASPQ2'] > 0) | (field_stars['ASPQ3'] > 999) |
+                 (field_stars['VAR'] > -9999))
```

Comment:
If we move the plotting code to chandra_aca, do you think this particular set of bad_star criteria should go there too as a default filter?  Or should this stay in starcheck?  My thought was that this "make_plots_for_obsid" routine would stay in starcheck.